### DISCUSSION TITLE
dynamodbattribute.UnmarshalMap for SDK v2

### DISCUSSION THREAD

**Discussion by leafypout** [aws/aws-sdk-go-v2#2169] [Q&A]
Is there a `dynamodbattribute.UnmarshalMap` function (or similar) for version 2 of the SDK? I'm currently having to use aws-sdk-go instead of aws-sdk-go-v2 for DynamoDB because the `dynamodbattribute.UnmarshalMap` function takes a `map[string]*dynamodb.AttributeValue` as input, but in SDK v2 the DynamoDB API calls (like `QueryOutput` and `ScanOutput`) return a `[]map[string]dynamodb.AttributeValue`.

(I know both return slices, but I'm specifically talking about how aws-sdk-go returns a pointer to AttributeValue, and aws-sdk-go-v2 doesn't).

**Comment by jamessouth** [comment]
It's [here](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue#UnmarshalMap). Also see [this discussion](https://github.com/aws/aws-sdk-go-v2/discussions/1652) and [this gist](https://gist.github.com/Luke-Davies/eb67ddd90f73676a64ce5b0c7cf2ba80), and then [here's a file of mine](https://github.com/jamessouth/aws-clean-tablet/blob/main/src/lambda/games/main.go) using the functions in the gist, SDK v2, UnmarshalMap, Query, etc.

**Comment by t0mpl** [comment]
I wonder if AWS team realises that their current implementation is absolutely unusable as is and if after 6 years they will start to consider a decent solution.
